metrics/collector.go

```diff
@@ -34,6 +34,7 @@ type Metric struct {
 type Collector struct {
 	metrics map[string]Metric
 	model   *sabakan.Model
+	mu      *sync.RWMutex
```

Comment:
Why do you use `sync.RWMutex` instead of `sync.Mutex`?
Since you are not using `RLock()`, `sync.Mutex` should be sufficient.